### PR TITLE
feat: refactor surroundings fom and dom [132]

### PR DESCRIPTION
**Pull request by Carsopre**
## Issue addressed
Solves #132

## Code of conduct
- [x] I HAVE NOT added sensitive or compromised (test) data to the repository.
- [x] I HAVE NOT added vulnerabilities to the repository.
- [x] I HAVE discussed my solution with (other) members of the KOSWAT team.

## What has been done?
- Created new surroundings  subtypes (`SurroundingsObstacle` and `SurroundingsInfrastructure`) to associate to each of the surroundings property in the `SurroundingsWrapper`.
- Removed irrelevant builders / classes.
- Modified `PointSurroundings` so that contains a matrix of distances and their 'surrounding' value at said distance rather than just a list of distances.
- Updated readers / importers to load all the csv data correctly.

### Checklist
- [x] Tests are either added or updated.
- [x] Branch is up to date with `master`.
- [x] Updated documentation if needed.

## Additional Notes (optional)
- Updated `README.md` as well as the `surroundings.md` documentation.
- Updated / extended tests.
- Renamed some files / classes to be more consistent with the rest of the code.

### CODE REVIEW

**Review by ArdtK — APPROVED** [review]
Some small remarks. Didn't go throught all the tests in detail as I don't have the time now.

File: koswat/configuration/io/csv/koswat_surroundings_csv_fom.py
[the remark this answers is not in the document]
Comment (Carsopre) [reply koswat/configuration/io/csv/koswat_surroundings_csv_fom.py:1]: Ah, I knew I was forgetting one but could not find it with my own eyes. Thanks!

[the remark this answers is not in the document]
Comment (Carsopre) [reply koswat/configuration/io/csv/koswat_surroundings_csv_fom.py:1]: wait this file should have not been added, I believe there's something wrong with the merges.

File: koswat/dike/surroundings/README.md [hunk 1]
```diff
@@ -1,4 +1,10 @@
...
 
 This module aims to contain all classes that represent the surroundings of a dike (or list of them).
-For instance, the `KoswatCsvFom` and `KoswatShpModel` will be represented here with a `DataObjectModel`.
\ No newline at end of file
+
+A "surrounding" is defined by the `KoswatSurroundingsProtocol`, it contains a list of points (`PointSurroundings`) which represent locations (`PointSurroundings.location`) in a dike traject and their distance(s) (`PointSurroundings.surroundings_matrix`) to a specific type of surrounding. We mainly distinguish two types of surroundings:
+
+- Obstacles (`SurroundingsObstacle`), they cannot be repaired or replaced (removed and built again), becoming a boundary for a `ReinforcementProfile` and therefore limiting which type of reinforcements can be applied at a given location (`PointSurroundings`).
+- Infrastructures (`SurroundingsInfrastructure`), they can be both reworked and / or replaced (removed and built again). When a `ReinforcementProfile` at a given location (`PointSurroundings`) has infrastructures in their `surroundings_matrix` then we use them to estimate additional costs depending on whether they will be reworked or replaced.
```
[the remark this answers is not in the document]
Comment (Carsopre) [reply koswat/dike/surroundings/README.md:8]: Funny enough in the line above I use indeed `repaired`

File: koswat/dike/surroundings/surroundings_obstacle.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+@dataclass
+class SurroundingsObstacle(KoswatSurroundingsProtocol):
+    """
+    Defines surroundings point collections that cannot be removed.
+    """
+
+    points: list[PointSurroundings] = field(default_factory=lambda: [])
+
+    def get_locations_after_distance(self, distance: float) -> list[Point]:
```
[the remark this answers is not in the document]
Comment (Carsopre) [reply koswat/dike/surroundings/surroundings_obstacle.py:20]: I think it does, I will address later this problem as I believe it should be used from the wrapper (which implies I need to rework on it).

File: koswat/dike/surroundings/point/point_surroundings.py [hunk 3]
```diff
@@ -13,7 +13,7 @@ class PointSurroundings:
     section: str = ""
     traject_order: int = -1
     location: Point | None = None
-    distance_to_surroundings: list[float] = field(default_factory=lambda: [])
+    surroundings_matrix: dict[float, float] = field(default_factory=dict)
```
[the remark this answers is not in the document]
Comment (Carsopre) [reply koswat/dike/surroundings/point/point_surroundings.py:16]: Not here, it should be described at the classes that use said property (thus `SurroundingsObstacle` and `SurroundingsInfrastructure`

File: koswat/dike/surroundings/point/point_surroundings.py [hunk 4]
```diff
@@ -44,9 +44,18 @@ def __eq__(self, __value: object) -> bool:
     @property
     def closest_surrounding(self) -> float:
         """
-        Distance to the closest surrounding (building/railway/water). When no surroundings are given the value will be `NaN` (Not A Number), so that the value 0 is reserved for buildings at distance 0.
+        Distance to the closest (obstacle) surrounding. When no surroundings are given the value will be `NaN` (Not A Number), so that the value 0 is reserved for buildings at distance 0.
```
[the remark this answers is not in the document]
Comment (Carsopre) [reply koswat/dike/surroundings/point/point_surroundings.py:47]: yes, good one.

File: koswat/configuration/io/csv/koswat_surroundings_csv_fom.py
[the remark this answers is not in the document]
Comment (Carsopre) [reply koswat/configuration/io/csv/koswat_surroundings_csv_fom.py:1]: Renamed classes.

**Pull request merged**